bitnami/argo-cd/templates/application-controller/metrics-svc.yaml

```diff
@@ -20,15 +20,24 @@ metadata:
   {{- end }}
 spec:
   type: {{ .Values.controller.metrics.service.type }}
+  {{- if and .Values.controller.metrics.service.clusterIP (eq .Values.controller.metrics.service.type "ClusterIP") }}
+  clusterIP: {{ .Values.controller.metrics.service.clusterIP }}
+  {{- end }}
   {{- if (or (eq .Values.controller.metrics.service.type "LoadBalancer") (eq .Values.controller.metrics.service.type "NodePort")) }}
   externalTrafficPolicy: {{ .Values.controller.metrics.service.externalTrafficPolicy | quote }}
   {{- end }}
-  {{ if eq .Values.controller.metrics.service.type "LoadBalancer" }}
+  {{- if eq .Values.controller.metrics.service.type "LoadBalancer" }}
   loadBalancerSourceRanges: {{ .Values.controller.metrics.service.loadBalancerSourceRanges }}
-  {{ end }}
+  {{- end }}
   {{- if (and (eq .Values.controller.metrics.service.type "LoadBalancer") (not (empty .Values.controller.metrics.service.loadBalancerIP))) }}
   loadBalancerIP: {{ .Values.controller.metrics.service.loadBalancerIP }}
   {{- end }}
+  {{- if .Values.controller.metrics.service.sessionAffinity }}
+  sessionAffinity: {{ .Values.controller.metrics.service.sessionAffinity }}
+  {{- end }}
+  {{- if .Values.controller.metrics.service.sessionAffinityConfig }}
+  sessionAffinityConfig: {{- include "common.tplvalues.render" (dict "value" .Values.controller.metrics.service.sessionAffinityConfig "context" $) | nindent 4 }}
+  {{- end }}
   ports:
     - name: metrics
       port: {{ .Values.controller.metrics.service.port }}
```

Comment:
This should be ports.metrics, right?

Reply:
I haven't standardized `.Values.controller.metrics.service.port` since it does not give us much value and it could be repetitive.